osquery/tables/system/linux/disk_encryption.cpp

```diff
@@ -0,0 +1,100 @@
+#include <unistd.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/sql.h>
+
+extern "C" {
+#include <libcryptsetup.h>
+
+crypt_status_info crypt_status(struct crypt_device *cd, const char *name);
+
+#ifdef CENTOS_CENTOS6
+int crypt_init_by_name(struct crypt_device **cd, const char *name);
+#define crypt_init(cd, name) crypt_init_by_name(&cd, name.c_str())
+#else
+int crypt_init_by_name_and_header(struct crypt_device **cd,
+				  const char *name,
+				  const char *header_device);
+#define crypt_init(cd, name) \
+  crypt_init_by_name_and_header(&cd, name.c_str(), opt_header_device)
+#endif
+
+const char *crypt_get_type(struct crypt_device *cd);
+
+int crypt_get_active_device(struct crypt_device *cd,
+			    const char *name,
+			    struct crypt_active_device *cad);
+
+const char *crypt_get_cipher(struct crypt_device *cd);
+
+const char *crypt_get_cipher_mode(struct crypt_device *cd);
+
+void crypt_free(struct crypt_device *cd);
+}
+
+namespace osquery {
+namespace tables {
+
+void genFDEStatusForBlockDevice(const std::string &name,
+				const std::string &uuid,
+				QueryData &results) {
+  Row r;
+  r["name"] = name;
+  r["uuid"] = uuid;
+
+  struct crypt_device *cd = nullptr;
+  const char *opt_header_device = nullptr;
+  struct crypt_active_device cad;
+  crypt_status_info ci;
+  std::string type;
+  std::string cipher;
+  std::string cipher_mode;
+
+  ci = crypt_status(cd, name.c_str());
+  switch (ci) {
+  case CRYPT_ACTIVE:
+  case CRYPT_BUSY:
+    r["encrypted"] = "1";
+
+    if (crypt_init(cd, name) < 0) {
+      VLOG(1) << "Unable to initialize crypt device for " << name;
+      crypt_free(cd);
+      break;
+    }
+    type = crypt_get_type(cd);
+    if (crypt_get_active_device(cd, name.c_str(), &cad) < 0) {
+      VLOG(1) << "Unable to get active device for " << name;
+      crypt_free(cd);
+      break;
+    }
+    cipher = crypt_get_cipher(cd);
+    cipher_mode = crypt_get_cipher_mode(cd);
+    r["type"] = type + "-" + cipher + "-" + cipher_mode;
+    break;
+  default:
+    r["encrypted"] = "0";
+  }
+
+  results.push_back(r);
+}
+
+QueryData genFDEStatus(QueryContext &context) {
+  QueryData results;
+
+  if (getuid() || geteuid()) {
+    VLOG(1) << "Disk encryption status not available. Are you running as root?";
```

Comment:
Do you mean keep the message declarative (remove the question) or is there a more declarative way of calling the logger?